routes/author/index.js

```diff
@@ -19,59 +18,64 @@ module.exports = function(req, res){
       } finally {
         book.images = images;
       }
-      if(err){
+      if (err) {
         console.log(err);
         return book;
       }
       return book;
     });
-  };
+  }
 
-  var authorId = Number(req.params[0]);
-  var authors = {};
+  const authorId = Number(req.params[0]);
+  const authors = {};
+  if (isNaN(authorId)) {
+    return res.status(404).render('error', {
+      title: 'ページが見つかりません。',
+    });
+  }
 
-  var countAuthors = function(){
-    var d = Q.defer();
-    Author.count({}, function(err, count){
+  function countAuthors() {
+    const deffered = Q.defer();
+    Author.count({}, (err, count)=> {
       authors.count = count;
-      d.resolve(authors);
+      deffered.resolve(authors);
     });
-    return d.promise;
-  };
+    return deffered.promise;
+  }
 
-  var getAuthor = function(authorId, index){
-    var d = Q.defer();
-    Author.findOne({ pageId: authorId }, function(err, author){
-      if(err){
-        return console.log("err", err);
+  function getAuthor(id, index) {
+    const deffered = Q.defer();
```

Comment:
__[jshint]__
*  'const' is available in ES6 (use esnext option) or Mozilla JS extensions (use moz).